### PR TITLE
Fix unhandled stream error in pipe

### LINKED ISSUE #614: Unhandled stream error in pipe caused by ECONNRESET
When I use `form-data` to upload file, in some complicate environments, the code will crash. Below error message will display.

```
internal/streams/legacy.js:59
      throw er; // Unhandled stream error in pipe.
      ^
Error: read ECONNRESET
    at TCP.onread (net.js:622:25)
```

### PR DESCRIPTION
**Pull request by conechan**
Fix #614

### CODE REVIEW

File: source/as-promise.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ module.exports = options => {
 
 			if (is.nodeStream(options.body)) {
 				options.body.once('end', uploadComplete);
+				request.on('error', error => reject(error));
```
Comment (szmarczak) [inline source/as-promise.js]: This should be before that `if`.

**Comment by sindresorhus** [comment]
@conechan Ping :)

**Comment by conechan** [comment]
Sorry, I've tried to write a test to reproduce it but failed. I will try again when I have spare time. Thanks.

**Comment by szmarczak** [comment]
I'll try to fix this :)

**Pull request closed without merging**